src/Microsoft.TemplateEngine.Cli/TableOutput/TemplateGroupDisplay.cs

```diff
@@ -0,0 +1,96 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.TemplateEngine.Abstractions;
+using Microsoft.TemplateEngine.Edge.Template;
+using Microsoft.TemplateEngine.Utils;
+
+namespace Microsoft.TemplateEngine.Cli.TableOutput
+{
+    internal static class TemplateGroupDisplay
+    {
+        /// <summary>
+        /// Generates the list of template groups for table display.
+        /// Except where noted, the values are taken from the highest-precedence template in the group. The info could vary among the templates in the group, but shouldn't. (There is no check that the info doesn't vary.)
+        /// - Template Name
+        /// - Short Name: displays the first short name from the highest precedence template in the group.
+        /// - Language: All languages supported by any template in the group are displayed, with the default language in brackets, e.g.: [C#]
+        /// - Tags
+        /// - Author
+        /// - Type
+        /// </summary>
+        /// <param name="templateList">list of templates to be displayed</param>
+        /// <param name="language">language from the command input</param>
+        /// <param name="defaultLanguage">default language</param>
+        /// <returns></returns>
+        internal static IReadOnlyList<TemplateGroupTableRow> GetTemplateGroupsForListDisplay(IReadOnlyCollection<ITemplateMatchInfo> templateList, string language, string defaultLanguage)
+        {
+            List<TemplateGroupTableRow> templateGroupsForDisplay = new List<TemplateGroupTableRow>();
+            IEnumerable<IGrouping<string, ITemplateMatchInfo>> groupedTemplateList = templateList.GroupBy(x => x.Info.GroupIdentity, x => !string.IsNullOrEmpty(x.Info.GroupIdentity));
+            foreach (IGrouping<string, ITemplateMatchInfo> grouping in groupedTemplateList)
+            {
+                List<string> languageForDisplay = new List<string>();
+                HashSet<string> uniqueLanguages = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+                string defaultLanguageDisplay = string.Empty;
+
+                foreach (ITemplateMatchInfo template in grouping)
+                {
+                    if (template.Info.Tags != null && template.Info.Tags.TryGetValue("language", out ICacheTag languageTag))
```

Comment:
Nit: reverting this `if` to `continue` immediately could reduce nesting and improve readability. Same goes for the first `if` in the `foreach` below.